README.md

```diff
@@ -189,28 +190,156 @@ When Silead touchscreen support was added to ACPI (PC) platforms, the
+
+Run `sudo evtest`. This gives a list of `/dev/input/event#` devices.
+Replace `#` in the commands below with the number of the `/dev/input/event#`
+node for the `silead_ts` input device.
+
+Run `sudo evtest /dev/input/event# | grep ABS_X` and then slowly run
+your finger over the screen from the left edge of the screen to
+the right edge along the middle line of the display and press Ctrl + C
+to terminate evtest when done. This should print slowly increasing or
+decreasing ABS_X values.
+
+If instead the ABS_X values stay approx. the same then your X and Y axis are
+swapped. Redo the test moving your finger over the screen from the top edge
+of the screen to the bottom edge along the middle line of the display instead.
+
+If there are big jumps or holes in the range, retry making sure your finger
+makes proper contact with the screen if that does not help then you have
+the wrong firmware.
+
+After verifying the X axis works well repeat the process for ABS_Y and
+verify that the Y axis works well too.
+
+#### 4. Determine the min and max values of the axis
+
+Run `sudo evtest /dev/input/event# | grep ABS_X` and for the same edges
+as used for ABS_X during step 3 first place your finger on the display
+near the edge and then *slowly* move it over the edge to the bezel,
+notice the lowest or highest value reported for ABS_X while doing this.
+Repeat this about 5 times to make sure you really have found
+the lowest / highest value the axis reports.
+
+Write down the lowest ABS_X value observed as x-min and the highest
+observed value (on the other edge) as x-max.
+
+Repeat this for ABS_Y and write down y-min and y-max.
+
+#### 5. Testing the parameters
+
+When running kernel 6.10 or later you can now test the found min/max
+values on the kernel commandline.
+
+This requires the ACPI Hardware ID (HID) for the touchscreen. Run
+`ls -d /sys/bus/i2c/devices/i2c-MSSL*`, this should output a single line
+ending in e.g. `i2c-MSSL1680:00`. The ACPI HID of the touchscreen is the bit
+after the `i2c-` and before the `:00` so `MSSL1680` in this case.
+
+This HID is the first argument for the `i2c_touchscreen_props` kernel
+commandline argument. To pass the found min/max values the following syntax
+is used:
+
+```
+i2c_touchscreen_props=<ACPI_HW_ID>:touchscreen-min-x=<x-min>:touchscreen-min-y=<y-min>:touchscreen-size-x=<x-max + 1>:touchscreen-size-y=<y-max + 1>
+```
+
+So if for example in step 4. you have found values of x-min=8, y-min=8,
+x-max=1919, y-max=1079 then the kernel commandline argument to pass would be:
+
+```
+i2c_touchscreen_props=MSSL1680:touchscreen-min-x=8:touchscreen-min-y=8:touchscreen-size-x=1920:touchscreen-size-y=1080
+```
+
+Edit your kernel configuration, e.g. `/boot/loader/entries/<current-kernel>.conf`,
+add this to your kernel commandline then reboot to test. After reboot run
+`sudo cat /proc/cmdline` to make sure that `i2c_touchscreen_props=...` is there
+and is correct.
+
```

Comment:
I'd move this part after the next paragraph. You already mentioned how to detect swapped axes, so adding the correct flags should be straight-forward.

Reply:
> I'd move this part after the next paragraph. You already mentioned how to detect swapped axes, so adding the correct flags should be straight-forward.

I did the split of first testing with only min/max settings vs adding swapping deliberately. The text to test the axis talks about top/bottom and left/right edges. Depending on form factor users are likely to define which edges are top/bottom and what is left/right from e.g. holding the device in landscape orientation. But the actual LCD panel could be a portrait panel.

So even though the user swiped from top to bottom when testing the X-axis, which feels like swapping is needed, swapping might not be necessary (and actually would be wrong to do) because they were holding the LCD panel 90° rotated from its native panel orientation.

So I prefer to keep this as is and first have users test with only min/max values set.